projects/rx-form-mapper/src/lib/services/form-reader.ts

```diff
@@ -0,0 +1,78 @@
+import { InjectFlags, Injector, Type } from '@angular/core';
+import { AbstractControl, FormArray, FormControl, FormGroup } from '@angular/forms';
+import { CustomControlMapper } from '..';
+import { ControlVisitor, CustomControlMetadata, FormArrayMetadata, FormControlMetadata, FormGroupMetadata, FormMetadata } from '../metadata';
+
+export class FormReader implements ControlVisitor<any> {
+
+	public constructor(private readonly control: AbstractControl, private readonly injector: Injector) {}
+
+	public visitCustomControlMetadata(customControlMetadata: CustomControlMetadata): any {
+		return this.resolveCustomMapper(customControlMetadata.mapper).readForm(this.control);
+	}
+
+	public visitFormArrayMetadata(formArrayMetadata: FormArrayMetadata) {
+		if (!this.control) {
+			return void 0;
```

Comment:
void 0?

Reply:
sì 😍 lo uso per 2 motivi:
1. void 0 costa meno caratteri di "undefined"
2. void 0 produce il valore undefined,  undefined è una variabile globale che contiene un valore di tipo undefined ma:
 a. in browser vecchi può essere sovrascritta
 b. puoi creare variabili con il nome undefined in un tuo scope locale `(function() { var undefined = 'ciao'; console.log(undefined);})(); // 'ciao'`

Reply:
Secondo me non è parlante a sufficienza. Se vuoi usare questa feature, essendo che void accetta una expression, metti quantomeno lo zero tra parentesi tonde.

In questo modo chiunque, soprattutto chi ha un passato in c/c++, sa riconoscere l'espressione e il valore che ne assume.

Reply:
corretto!